src/main/java/net/neoforged/neoforge/attachment/IAttachmentCloner.java

```diff
@@ -0,0 +1,12 @@
+package net.neoforged.neoforge.attachment;
+
+import javax.annotation.Nullable;
+
+/**
+ * Custom cloner for data attachments, to improve efficiency compared to the default
+ * serialize-copy nbt-deserialize-implementation.
+ */
+public interface IAttachmentCloner<T> {
+    @Nullable
+    T copy(IAttachmentHolder holder, T attachment);
```

Comment:
A bit annoying that the method and interface names don't match. However `Copier` is really ugly, and `clone` as a method name is not as nice either.

Reply:
yeah, I didn't like Copier either, but maybe it's better to have it all match, because everything else uses copy

Reply:
Can always do `IAttachmentCopyHandler`.